### ISSUE TITLE
🐛 [BUG] Segmentation fault (core dumped) when running on large systems on LAMMPS

### ISSUE THREAD

**Issue by k50112113** [mir-group/pair_nequip#38]
**Describe the bug**
I have successfully trained my water model with 64-water system and it works fine when I run LAMMPS upto 1728 H2O, but when I went to 4096 H2O, it gave "Segmentation fault (core dumped)". Is there a limitation on how many atoms NequIP can handle?

**To Reproduce**
units metal
atom_style atomic
boundary p p p
newton off
neighbor 1.0 bin
neigh_modify every 1 delay 0 check yes
read_data water-64.data
pair_style	nequip
pair_coeff	* * base-float32.pth H O
replicate 4 4 4
timestep 0.001
thermos 1
dump        dump1 all custom 10 traj.lammpstrj id type element x y z 
dump_modify dump1 sort id element H O
velocity all create 300.0 24829580
thermo_style custom step temp press density pe
fix nvt all nvt temp 300.0 300.0 $(100.0*dt)
run 100

**Environment (please complete the following information):**
 - OS: Ubuntu, Windows
 - LAMMPS: (29 Sep 2021 - Update 2)
 - nequip_version: 0.5.6
 - e3nn_version: 0.5.1
 - pytorch version: 1.11.0
 - cuda Version: 10.2

**Additional context**
I can sent my deployed model if anyone can help address this issue. 
Some important information about my config.yaml:
default_dtype: float32
r_max: 4.0
num_layers: 3
l_max: 1
parity: false
num_features: 16
nonlinearity_type: gate 
resnet: false 
nonlinearity_scalars:
  e: silu
  o: tanh
nonlinearity_gates:
  e: silu
  o: tanh
num_basis: 16
BesselBasis_trainable: true
PolynomialCutoff_p: 6
invariant_layers: 2
invariant_neurons: 64
avg_num_neighbors: auto
use_sc: true
dataset: npz
key_mapping:
  z: atomic_numbers
  E: total_energy
  F: forces
  R: pos                                                                           
  cell: cell
  pbc: pbc
npz_fixed_field_keys:
  - atomic_numbers
  - pbc
chemical_symbols:
  - O
  - H
wandb: false
n_train: 4400              
n_val: 100               
learning_rate: 0.001
batch_size: 5
validation_batch_size: 5
max_epochs: 350
train_val_split: random
shuffle: true
metrics_key: validation_loss
use_ema: true
ema_decay: 0.99
ema_use_num_updates: true
report_init_validation: true
loss_coeffs:
  forces: 10
  total_energy:                                                                    
    - 1
    - PerAtomMSELoss
optimizer_name: Adam
lr_scheduler_name: ReduceLROnPlateau
lr_scheduler_patience: 100
lr_scheduler_factor: 0.5

**Comment by Linux-cpp-lisp** [comment]
Hi @k50112113 ,

Thanks for your interest in our code.

Please see https://github.com/mir-group/pair_nequip/issues/31.
